packages/babel-generator/src/generators/types.ts

```diff
@@ -240,16 +240,14 @@ export function DecimalLiteral(this: Printer, node: t.DecimalLiteral) {
 // Hack pipe operator
 export function TopicReference(this: Printer) {
   const { topicToken } = this.format;
-  switch (topicToken) {
-    case "#":
-      this.token("#");
-      break;
-
-    default: {
-      const givenTopicTokenJSON = JSON.stringify(topicToken);
-      const message = `The "topicToken" generator option must be "#" (${givenTopicTokenJSON} received instead).`;
-      throw new Error(message);
-    }
+  const validTopicTokenSet = new Set(["^", "%", "#"]);
```

Comment:
Nit: this can be moved to outside from the function.

Reply:
Resolved in 6b1a563.